### ISSUE TITLE
Environment depth of field far blur should have the option to not affect canvas backgrounds

### ISSUE THREAD

**Issue by elvisish** [godotengine/godot-proposals#3462]
### Describe the project you are working on

3D characters on 2D canvas background.

### Describe the problem or limitation you are having in your project

The depth of field affects canvas backgrounds when using Canvas mode in WorldEnvironment, there should be an option to have 3D elements affected by DOF but not the canvas background.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Have a checkbox to ensure the DOF stops when it goes back as far as the canvaslayer background, or have it rendered in a seperate pass.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

A simple checkbox to allow canvaslayer backgrounds to not be blurred by depth of field.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not that I know of.

### Is there a reason why this should be core and not an add-on in the asset library?

It would need to be integrated with WorldEnvironment.

**Comment by Calinou** [comment]
@elvisish Can you create a minimal reproduction project for this?

PS: The proposal's title seems to contradict the body.

**Comment by elvisish** [comment]
> @elvisish Can you create a minimal reproduction project for this and link it here?
> 
> 
> 
> PS: The proposal's title seems to contradict the body. Shouldn't the title read *"Enviroment Dof far blur should have the option to **not** affect canvas backgrounds"*?

You're right, I don't know why I phrased it like that. As the proposal describes, there should be an option to allow DOF on the camera, but have the canvas background fully in focus so it's not affected by the DOF at all.
